moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_states.cpp

```diff
@@ -249,4 +259,169 @@ void MotionPlanningFrame::clearStatesButtonClicked()
   populateRobotStatesList();
 }
 
+void MotionPlanningFrame::planAllStatesButtonClicked()
+{
+#if 0
+  planning_display_->addBackgroundJob(boost::bind(&MotionPlanningFrame::planAllStatesCartesianExec, this),
+                                      "plan all cartesian states");
+#else
+  planning_display_->spawnBackgroundJob(boost::bind(&MotionPlanningFrame::planAllStatesJointSpaceExec, this));
+#endif
+}
+
+void MotionPlanningFrame::planAndExecuteAllStatesButtonClicked()
+{
+#if 0
+  planning_display_->addBackgroundJob(boost::bind(&MotionPlanningFrame::planAndExecuteAllStatesCartesianExec, this),
+                                      "plan and execute all cartesian states");
+#else
+  planning_display_->spawnBackgroundJob(boost::bind(&MotionPlanningFrame::planAndExecuteAllStatesJointSpaceExec, this));
+#endif
+}
+
+void MotionPlanningFrame::planAllStatesCartesianTrajectory()
+{
+  std::vector<geometry_msgs::Pose> waypoints;
+  for (RobotStateMap::iterator it = robot_states_.begin(); it != robot_states_.end(); ++it)
+  {
+    robot_state::RobotState robot_state(*planning_display_->getQueryGoalState());
```

Comment:
This is odd because most of the Rviz moveit plugin uses MoveGroup to do planning "over the wire", but here you're doing direct motion planning using low level MoveIt plugins. This mix makes me uncomfortable.